content/learn/book/migration-guides/0.4-0.5/_index.md

```diff
@@ -0,0 +1,69 @@
++++
+title = "0.4 to 0.5"
+weight = 1
+sort_by = "weight"
+template = "book-section.html"
+page_template = "book-section.html"
+[extra]
+long_title = "Migration Guide: 0.4 to 0.5"
++++
+
+<!-- TODO: link to release blog post here -->
+
+## `commands: &mut Commands` SystemParam is now `mut commands: Commands`
+
+```rust
+// 0.4
+fn foo(commands: &mut Commands) {
+}
+
+// 0.5
+fn foo(mut commands: Commands) {
+}
+```
+
+Systems using the old `commands: &mut Commands` syntax in 0.5 will fail to compile when calling `foo.system()`.
+
+This change was made because `Commands` now holds an internal `World` reference to enable safe Entity allocations.
+
+Note: The internal `World` reference requires two lifetime parameters to pass Commands into a non-system function: ```commands: &'a mut Commands<'b>```
+
+## Systems allow a maximum of 12 top-level SystemParams, down from 15
+
+```rust
+// 0.4
+fn foo(r1: Res<Thing1>, r2: Res<Thing2>, r3: Res<Thing3>, r4: Res<Thing4>, ... r15: Res<Thing15>) {
+}
+
+// 0.5
+fn foo(r1_thru_3: (Res<Thing1>, Res<Thing2>, Res<Thing3>), r4: Res<Thing4>, ... r15: Res<Thing15>) {
```

Comment:
```suggestion
fn foo(r1_thru_3: (Res<Thing1>, Res<Thing2>, Res<Thing3>), r4: Res<Thing4>, ... r12: Res<Thing12>) {
```

Reply:
oh, I was too quick to comment. I see now how its using the nested for the first three ^^

Reply:
Maybe it would be good to pattern match the first argument to help with noticing the change?

```suggestion
fn foo((r1, r2, r3): (Res<Thing1>, Res<Thing2>, Res<Thing3>), r4: Res<Thing4>, ... r15: Res<Thing15>) {
```